src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions.cs

```diff
@@ -190,11 +190,20 @@ private static SyntaxNode CreateNewArgumentNullException(SyntaxGenerator factory
                 : factory.ReferenceEqualsExpression(identifier, nullExpr);
 
             // generates: if (s == null) throw new ArgumentNullException(nameof(s))
-            return factory.IfStatement(
+            var throwStatement = factory.ThrowStatement(CreateNewArgumentNullException(
+                                    factory, semanticModel.Compilation, parameter));
+            throwStatement = throwStatement.WithLeadingTrivia(factory.Whitespace(" ")).WithoutAnnotations(SyntaxAnnotation.ElasticAnnotation);
+
+            var ifStatement = factory.IfStatement(
                condition,
                 SpecializedCollections.SingletonEnumerable(
-                    factory.ThrowStatement(CreateNewArgumentNullException(
-                        factory, semanticModel.Compilation, parameter))));
+                    throwStatement));
+
+            var close = ifStatement.FindToken(10);
+            var newClose = close.WithoutTrailingTrivia();
+            ifStatement = ifStatement.ReplaceToken(close, newClose);
+
+            return ifStatement;
```

Comment:
Both, the `throw` keywords leading trivia and the `)` of the if condition need their respective trivia `SyntaxAnnotation.ElasticAnnotation` removed.